Mathlib/Order/Restriction.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2024 Etienne Marion. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Etienne Marion
+-/
+import Mathlib.Order.Interval.Set.Basic
+import Mathlib.Order.Interval.Finset.Basic
+
+/-!
+# Restriction of a function indexed by natural integers
+
+Given a preorder `α` and dependent function `f : (i : α) → β i` and `a : α`, one might want
+to consider the restriction of `f` to elements `≤ a`.
+This is defined in this file as `Prerder.restrict a f`.
+Similarly, if we have `a b : α`, `hab : a ≤ b` and `f : (i : ↑(Set.Iic b)) → β ↑i`,
+one might want to restrict it to elements `≤ a`.
+This is defined in this file as `Preorder.restrict₂ hab f`.
+
+We also provide versions where the intervals are seen as finite sets, see `Preorder.frestrict`
+and `Preorder.frestrict₂`.
+
+## Main definitions
+* `Preorder.restrict a f`: Restricts the function `f` to the variables indexed by elements `≤ a`.
+-/
+
+namespace Preorder
+
+variable {α : Type*} [Preorder α] {β : α → Type*}
```

Comment:
Could you use `π` instead of `β`, to make sure that the names are the same between the different restriction variants? This is important when you name the implicit argument.